app/(main)/account/recovery/page.tsx

```diff
@@ -0,0 +1,263 @@
+ */
+const ResetPasswordSchema = z
+  .object({
+    password: z
+      .string()
+      .min(1, { message: 'Please enter a password' })
+      .min(6, { message: 'Password must be at least 6 characters' }),
+    confirmPassword: z
+      .string()
+      .min(1, { message: 'Please confirm your password' })
+      .min(6, { message: 'Password must be at least 6 characters' }),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords don't match",
+    path: ['confirmPassword'],
+  });
+
+type ResetUserPasswordSchemaType = z.infer<typeof ResetPasswordSchema>;
+
+/**
+ * Renders the recover password page.
+ * @returns {React.JSX.Element} The rendered login page.
+ */
+const ResetPassword = (): React.JSX.Element => {
+  const router = useRouter();
+  const { isSignedIn, getUser } = useAuthContext();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  let userId: string | null = null;
+  let secret: string | null = null;
+  let expire: string | null = null;
+
+  /**
+   * Wrapper component to handle search params
+   * @returns {null} - null
+   */
+  const SearchParamsWrapper = (): null => {
+    const searchParams = useSearchParams();
+    userId = searchParams.get('userId');
+    secret = searchParams.get('secret');
+    expire = searchParams.get('expire');
+
+    useEffect(() => {
+      if (!userId || !secret || !expire) {
+        router.push('/login');
+        return;
+      }
+
+      setIsLoading(false);
+
+      if (isSignedIn) {
+        getUser();
+        router.push('/league/all');
+      }
+    }, [isSignedIn]);
+
+    return null;
+  };
+
+  const form = useForm<ResetUserPasswordSchemaType>({
+    resolver: zodResolver(ResetPasswordSchema),
+    defaultValues: {
+      password: '',
+      confirmPassword: '',
+    },
+  });
+
+  /**
+   * The current value of the 'password' field in the form.
+   * @type {string}
+   */
+  const password: string = useWatch({
+    control: form.control,
+    name: 'password',
+    defaultValue: '',
+  });
+
+  /**
+   * The current value of the 'confirmPassword' field in the form.
+   * @type {string}
+   */
+  const confirmPassword: string = useWatch({
+    control: form.control,
+    name: 'confirmPassword',
+    defaultValue: '',
+  });
+
+  /**
+   * Handles the form submission.
+   * @param data - The data from the form.
+   */
+  const onSubmit: SubmitHandler<ResetUserPasswordSchemaType> = async (data) => {
+    const { password } = data;
+    setIsSubmitting(true);
+
+    try {
+      await resetRecoveredPassword({
+        userId: userId as string,
+        token: secret as string,
+        password,
+      });
+
+      toast.custom(
+        <Alert
+          variant={AlertVariants.Success}
+          message="Your password has been reset successfully! Please login with your new password."
+        />,
+      );
+
+      setTimeout(() => router.push('/login'), 2000);
```

Comment:
Why are we setting a timeout? This should go right away.

Reply:
✅ fixed.